app/src/main/java/com/example/randomuser/viewmodel/UserViewModel.kt

```diff
@@ -0,0 +1,82 @@
+package com.example.randomuser.viewmodel
+
+import androidx.compose.runtime.*
+import androidx.lifecycle.ViewModel
+import androidx.lifecycle.viewModelScope
+import com.example.randomuser.data.Filter
+import com.example.randomuser.model.Results
+import com.example.randomuser.repo.Repo
+import kotlinx.coroutines.Dispatchers
+import kotlinx.coroutines.launch
+import com.example.randomuser.model.Result as Result
+
+class UserViewModel : ViewModel() {
+
+    var resultsResponse: Result? by mutableStateOf(null)
+    var visibleUsers = mutableStateListOf<Results>()
+    var activeFilters = mutableStateListOf<Filter>()
+
+    fun setup() {
+        viewModelScope.launch(Dispatchers.IO) {
+            val response = Repo().fetchAllUsersRetrofit()
+            resultsResponse = response
+            visibleUsers.addAll(response.results)
+        }
+    }
+
+    fun getFilteredUsers(selectedFilter: MutableState<Filter?>) {
+        val users = resultsResponse?.results
+        visibleUsers.clear()
+
+        if (selectedFilter.value?.selected == true) {
+            activeFilters.add(selectedFilter.value!!)
+        }
+
+        if (selectedFilter.value?.selected == false) {
+            activeFilters.remove(selectedFilter.value!!)
+        }
+
+        if (selectedFilter.value?.selected == false && selectedFilter.value?.selected == false) {
+            if (users != null) {
+                visibleUsers.addAll(users)
+            }
+        }
+
```

Comment:
You need a single list `response.results`

Look:
**Filter**
https://kotlinlang.org/docs/collection-filtering.html

**Remember the Order Rule**
https://kotlinlang.org/docs/collection-ordering.html

**Understand better Sequences**
https://kotlinlang.org/docs/sequences.html#sequence
https://kotlinlang.org/api/latest/jvm/stdlib/kotlin.sequences/distinct-by.html